src/Functions/FunctionsExternalDictionaries.h

```diff
@@ -594,31 +615,122 @@ class FunctionDictGetNoType final : public IFunction
 
 private:
 
+    std::pair<ColumnPtr, ColumnPtr> getDefaultsShortCircuit(
+        IColumn::Filter & default_mask,
```

Comment:
Yes, it it moved, but then moved back, and `default_mask` would be used later by function `restoreShortCircuitColumn`, so I think should still keep the same.

Reply:
Ah, yes, we moved it back, then it's ok. But to be honest, we do a lot of stuff with `default_mask` that sounds a bit unnecessary, like, move it around, reverse, create a column from it. Ideally we should be able to work with initial `default_mask` that was created by `getColumnsOrDefaultShortCircuit` and not modify it/create columns from it. If we change the code around `default_mask` so it initially contains `1` for rows where defaults should be used, I think we will be able to avoid all of this.

Reply:
After removing `expand`, it turns out passing `default_mask` by rvalue reference would be the best choice.